src/aeso_parser.erl

```diff
@@ -96,17 +96,29 @@ decl() ->
     choice(
       %% Contract declaration
     [ ?RULE(token(main), keyword(contract),
-            con(), tok('='), maybe_block(decl()), {contract_main, _2, _3, _5})
+            con(), tok('='), maybe_block(decl()), {contract_main, _2, _3, [], _5})
```

Comment:
This looks like an exponential blowup - we should consider a rewrite some day...

Reply:
@ghallak would you leave a TODO comment to verify it?

Reply:
or just check if it has a performance danger